File: easy/trick_or_treat.go
package main

import (
	"bufio"
	"fmt"
	"log"
	"os"
)

func main() {
	var v, z, w, h int
	data, err := os.Open(os.Args[1])
	if err != nil {
		log.Fatal(err)
	}
	defer data.Close()
	scanner := bufio.NewScanner(data)
	for scanner.Scan() {
		fmt.Sscanf(scanner.Text(),
			"Vampires: %d, Zombies: %d, Witches: %d, Houses: %d",
			&v, &z, &w, &h)
		fmt.Printf("%d\n", (v*3+z*4+w*5)*h/(v+z+w))
	}
}

File: easy/trick_or_treat.rb
File.open(ARGV[0]).each_line do |line|
  s = line.scan(/Vampires: (\d+), Zombies: (\d+), Witches: (\d+), Houses: (\d+)/)[0].map(&:to_i)
  puts (s[0] * 3 + s[1] * 4 + s[2] * 5) * s[3] / (s[0] + s[1] + s[2])
end
